torch/utils/tensorboard/_pytorch_graph.py

```diff
@@ -0,0 +1,267 @@
+import numpy as np
+import time
+from collections import OrderedDict
+
+from tensorboard.compat.proto.config_pb2 import RunMetadata
+from tensorboard.compat.proto.graph_pb2 import GraphDef
+from tensorboard.compat.proto.step_stats_pb2 import StepStats, DeviceStepStats, NodeExecStats, AllocatorMemoryUsed
+from tensorboard.compat.proto.versions_pb2 import VersionDef
+
+import torch
+from ._proto_graph import node_proto
+from torch.onnx.utils import OperatorExportTypes
+
+
+methods_OP = ['attributeNames', 'hasMultipleOutputs', 'hasUses', 'inputs',
+              'kind', 'outputs', 'outputsSize', 'scopeName']
+methods_IO = ['node', 'offset', 'uniqueName']  # 'unique' <int> , 'type' <Tensor<class 'torch._C.Type'>>
+
+
+class NodeBase(object):
+    def __init__(self, uniqueName=None, inputs=None, scope=None, tensor_size=None, op_type='UnSpecified', attributes=''):
+        self.uniqueName = uniqueName
+        self.inputs = inputs
+        self.tensor_size = tensor_size
+        self.kind = op_type
+        self.attributes = attributes
+        if scope is not None:
+            self.scope = scope
```

Comment:
I really want documentation for what each of these attributes mean.

Reply:
@lanpa can you comment here? Thanks.

Reply:
uniqueName: the unique name of the node e.g. `16` below
inputs: the input. e.g. the input of `19` is `16` and `18`
tensorSize: the size of each dimension. Retrieved by `node.type().sizes()`
kind: It will be shown in the detailed information pop-up in tensorboard. nodes of type Node_py_OP will get its kind by calling the `.kind()`
self.attributes = attributes e.g. `kernel_shape=[2, 2], pads=[0, 0, 0, 0], strides=[2, 2]` in `%15`. Retrieved by `str({k: Node_cpp[k] for k in Node_cpp.attributeNames()}).replace("'", ' ')`, and
`replace` is used to escape invalid chars in tensorboard


```
  %15 : Float(13, 10, 12, 12) = onnx::MaxPool[kernel_shape=[2, 2], pads=[0, 0, 0, 0], strides=[2, 2]](%14), scope: Net1
 %16 : Float(13, 10, 12, 12) = onnx::Relu(%15), scope: Net1
  %17 : Float(13, 10, 12, 12) = onnx::Neg(%15), scope: Net1
  %18 : Float(13, 10, 12, 12) = onnx::Relu(%17), scope: Net1
  %19 : Float(13, 10, 12, 12) = onnx::Add(%16, %18), scope: Net1
```